privacyidea/lib/tokens/webauthn.py

```diff
@@ -1167,6 +1168,45 @@ def verify_attestation_statement(fmt, att_stmt, auth_data, client_data_hash=b'',
                 #   * If successful, return attestation type ECDAA and
                 #     attestation trust path ecdaaKeyId.
                 raise RegistrationRejectedException('ECDAA attestation type is not currently supported.')
+            else:
+                # self-attestation
+                # Step 1:
+                # Validate that alg matches the algorithm of the credentialPublicKey
+                # in authenticatorData.
+                try:
+                    public_key_alg, credential_public_key = _load_cose_public_key(credential_pub_key)
+                except COSEKeyException as e:
+                    raise RegistrationRejectedException(str(e))
+                if alg != public_key_alg:
+                    raise RegistrationRejectedException('credentialPublicKey algorithm {0!s} does '
+                                                        'not match algorithm from attestation '
+                                                        'statement {1!s}'.format(public_key_alg, alg))
+                # Step 2:
+                # Verify that sig is a valid signature over the concatenation of authenticatorData
+                # and clientDataHash using the credential public key with alg.
+                try:
+                    _verify_signature(credential_public_key, alg, verification_data, signature)
+                except InvalidSignature:
+                    raise RegistrationRejectedException('Invalid signature received.')
+                except NotImplementedError:  # pragma: no cover
+                    # We do not support this. Treat as none attestation, if acceptable.
```

Comment:
Should we add a log entry here in info or warning.
Since we have an unsupported attestation - this might not be the expected behaviour.
Would it make sense to add a log, so that the admin realizes, that something is happening, that was not intended?